dvc/render/converter/vega.py

```diff
@@ -0,0 +1,316 @@
+
+    def _infer_fields(self):
+        fields = self.plot_properties.get("fields", set())
+        if fields:
+            fields = {
+                *fields,
+                self.plot_properties.get("x", None),
+                self.plot_properties.get("y", None),
+                self.inferred_properties.get("x", None),
+            } - {None}
+            self.inferred_properties["fields"] = fields
+
+    def _infer_y(self, datapoints: List[Dict]):
+        if "y" not in self.plot_properties:
+            data_fields = list(first(datapoints))
+            skip = (
+                REVISION_FIELD,
+                self.plot_properties.get("x", None)
+                or self.inferred_properties.get("x"),
+                FILENAME_FIELD,
+                VERSION_FIELD,
+            )
+            inferred_y = first(
+                f for f in reversed(data_fields) if f not in skip
+            )
+            if "y" in self.inferred_properties:
+                previous_y = self.inferred_properties["y"]
+                if previous_y != inferred_y:
+                    raise DvcException(
+                        f"Inferred y ('{inferred_y}' value does not match"
+                        f"previously matched one ('f{previous_y}')."
+                    )
+            else:
+                self.inferred_properties["y"] = inferred_y
+
+    def convert(
+        self,
+        data,
+        revision: str,
+        filename: str,
+        skip: List = None,
+        **kwargs,
+    ):
+        """
+        Convert the data. Fill necessary fields ('x', 'y') and return both
+        generated datapoints and updated properties.
+        """
+        if not skip:
+            skip = []
+
+        processed = deepcopy(data)
+
+        for step_name, step in self.steps:
+
+            if step_name not in skip:
+                processed = step(  # type: ignore
+                    processed,
+                    revision=revision,
+                    filename=filename,
+                )
+
+        self._infer_y(processed)  # type: ignore
+        return processed, {**self.plot_properties, **self.inferred_properties}
+
+    def _generate_y_values(  # noqa: C901
+        self,
+        datapoints: List[Dict],
+        y_values: Optional[Union[str, List, Dict]],
+        revision: str,
+        filename: str,
+        **kwargs,
+    ) -> List[Dict]:
+
+        result = []
+        properties_update = {}
+
+        def _add_version_info(datapoint, version_info):
+            tmp = datapoint.copy()
+            tmp[VERSION_FIELD] = version_info
+            tmp[REVISION_FIELD] = version_info["revision"]
+            return tmp
+
+        def _version_info(revision, filename=None, field=None):
+            res = {"revision": revision}
+            if filename is not None:
+                res["filename"] = filename
+            if field is not None:
+                res["field"] = field
+            return res
```

Comment:
@pared I don't fully understand the motivation for adding the `VERSION_FIELD` and the structure used. Could you clarify?

Reply:
`VERSION_FIELD` is intended as future replace for `rev` - it gives more information in a structured format. I presume (still need confirmation though) it will be used especially in `vscode` extension to have better control over plots there. One more perk is that providing this information in this format could allow us to deprecate `rev` (for ex in 3.0) and make our templates better utilize data version info. For example we could modify the plots to work in this way: [example](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUMSSCGcCGgDaoZcSgB9GrIBmcAE524y8wE9z23XDQBGAAy+lUwtSOwh2WUsbNjRQaxo4QWU0EBxBNkxzX0UQOME4WSQEL3RVdSkZeSVHYhowuWSACQBRAEEAERAAXyVc-MLilTUqaXqlVPTMn38xtIzvKe8Zib8fPyWMgCYpjfXzAGYpvd2AFinj3bEpsV2ANimb3YB2KcfdgA4pt92ATinvqrgxEarTaqFQpWGFTB4wyWSUAA80L5uiYzJFbA4nK53Do9AtAmiQnUIlZrNFNDl4olkjDzPMejQ8gUiskIeVRiBqrVwsD2l0GUz+qyhuzKilZpNUH4AuLlvjZXMslK1grzFspb4dqqDhqjqrThrzqrLhrrqq7hqHqrnhrXqqPhqvqrfhr-pzAbzQeCRSNZNCJfSQIjUMiFKiLKT7I5nG4PHiNVqgu57MT0eTYlSkuhaVsBX0WSUfRVslziZ7+ZTBQXBmVfdlaUrpbt5hrFqqVsqZTntrsdX49bSDX4jbSTX4zbSLX4rbSbX47bSHX4nbSXX43dVPWC2b7-RNc0GkSiVGjI5iYzjPIcCcEU+E0zFKQks9rsr1mQMd8WATUy+hmny3SVvmn5FhyDZTF2AbyrSHbeCq3YJr2176mcFxXLc9xPC87yfD8fw-lu3q1lCqC0gcCJHmGJ4RjYUZYrGuLFMON7JqE96kumT7UtmEqnHmH7CiRHKljy-4ghW75CoWwlihBGpQXKrbNo2CESuqfhauRKGDmhxoYeaWHWjh9p4c6BHukC4ntNuYF+mRfHZMGobhui9EXnGzG+GaSZEhxUSPnEz40hKlwCdJNaQiJgLcvU1kdEBUnVl+4ESo20ztjBAaqYpmw9q+uonHpo4GZORmziZi5mauFmbvFtmyXuGRhYeIbHkmZ7RtinlGb5d5yA+FJBTxhnhcldkljFf4gABCVjaBsn1mlkHNllyw5bsGmashhWoYa6GmphlrYbauGOvhrqEfVxFRfZZWUW11EdXR57dUxJl9exA2cYFmYhRMzzzUJt2Tb+YkzRJiWMiBwOiktEzpbldJrYqqxI1tWkSn2vgDo5+36YdhnHcZp2med5mXZZREpXdEqA61Lk0W5r2MVe5OfamP1DX9vETB8QMySDP6xfI8WSdDgmC3DuyI6tynthtqoYzt-ZFfjJWE2VxMVaTVXkzVlN1RDNk3aKTXmPzDPtaeL1daz8brqxfnfQF3PBbzGS-ALkXS5ZIvllDVYLULqqy5l8uwYriGaSrONqyxBPjkd04nfOZ3LhdjtU9dNPm17VtPTbGJ25eDsqhz-lkr97sK2+EsRTToP+2Lgcw1LdYyyt4d+G2kdo5t+Xabtunq6FpUSlOvgznTlUSkuvgrhKa6+BuHo53Z5srA9jPPcXDGl15bYVy7VduyNEHe43wvTbN4tB7DHehyjyNIbXyq9+pOlY8VeNJ4nU-J18AuGe+s54WQNvBGURtZoNVuubOErVvDW1onvDy71X7HxJK7DMNdYJ13vu3b8fsb6Q0vhNTuEdoKvz7u-Aew9v6jwmEObyB0U5EzTiTDOZMs4QJVNAkEsCzYOTlA9JBhcUHuTemzVWTNnZYNPjg8+n8yGLWvuDW+rdJY+0fvJHucsY5v3gh-fcX8JjYxHL-Fh-9p4TEnsAgG1UwGU14VAtexsvS52EXlURyDmYlx6vjTBg1FEvm0iokOok4ruLvm3bRRDdG+GMXMQe2VVhJLVKYjI5j45WI1mwrWHCdZcL1jwpe-ds7uMEbuLx+wnI+F8Z1feAS-5BK5iE-6GR+LAS0VfYh6jSHdIbuQp+lClIGJoUYuhMih4JxHn-PJAD2FANTgvdOK9M58LcTA021TdI+PEX4pp6C2GtOwdxUJY9wm+0iaLaJmihmqJGXo7u21DHwSmXHAqszLETjHtrCes8HGgL5uAspnYrqVJ2aREq+zXKNLQdI5Zsj+ryK4sNC5tj8GxN6TcgOVydHLVGck6hqTaFK0ybUxhnTx4TDHIsgpSKimrO4esimkCIXbM8fdRBDTbZHMRU6U5CjzkdPMPTJKwdrlTX6YBfF8TCXPLwSS9aaSPm4zMT-JhNLmrlQBbrEBxSnGlImGuTZVlIVcpnrCpm8KpEO1ZUKtFPNTJyuimDKJGjXVyQVYk-RryJnvPJfQjVVLzDMN+bS3VmL9VAsNSC5xYL2UVM5RvGpltgxiLhXyhFZcZSOurkok1WKenDNxS3L18NYTPxbOM0lkyg3TIYd8rVms-n5L1YUg1zKSmspcRygRUK5B5zqVKXlqC7XMUDfms+GLUYVrUR6gZEqH7yoRtWlJKqyXR0+TM3wFiW3zLbfSjtjKu2L3jcaz25T+Em0tSq61u9JH2y8omQkKLgkio9pMedfTF2ysGeNR5CTMZjJ3XW7wIG8qhuxhG6l-zI0xoyJPc9GQ5y9qNWpItZKb0eNTRfVqGwx1PoPihad7Sv2BmXYQt1zc7k-uA369VqNlSQYydB7VYao06sQ+YZDKy3QlMw1e7DWyB13uSQ9QjBzbXPr0mRz95Kf1lrowByVBK13KuJY2zd8FWNbX3VkjjzCbHcc7dGuNqHQUXqE+YU1bYcNVOhcogjRGWYkcJvJ9FoqKKqZXTRkh-6qNxNShp2toGmN0lUnpzVhnW0HuPQhszSHHGxvQ9ZpGdn+23rw9-STrn-HHMWZ551ezfPUbFMpz1ZXgvetC-6qh2nmO6Y+QZylh74smfMHS+xyXgWWYTeljZ9nRPZcajUrpwYpNZvHbJ0mxXcGXOqzi6Vf65pLeGQxl5EW4IQZa7HWDnH4OmdPeZ7tXaBMYYy6q5NYmcuRry9J7NE6zLzcLclpTK3blVaC70zbSrGuRdWNF9jcW4PtsSyd3rFmxVWYyEuGzmWbujbgTUu4D3pvEeaQ6t9X1UUFtnTDj77qvtLvroBkOf2GtgZ07thte79s5M691wBKGYcDbh+U3hw3zUprGxVdHNqnuzaTa9gnlsfuls+3i9bQGfWsZrdTprtPt2tZg4z1hkPeMpf65ei2nPE2Boc4O2mIKBePrc80qdOPOZnK81+-OEvHmVdJwQmrlbJRha04rwHLG9tfNyZYpngKodndSxdwbbLDcjdw3zspZui6Y+ObT0Xoqt4y4iVL8t6ffaU-CypIHfvd0HeMxrnrWu+ts91-Dq7Imee3dj-elzj2ZvubeHm63lcnULbXUT2js0AB+8x7nk5z4Stvcs3jpI7N8dGaA3isZ1G8CLBo3itZNG8A7Fo3idZtG8MvDo3is5dG8Vedf2iD8c0OmpCDnL5f5fGSfTt31tIU3g3vAW2iD5iSW2XGnx-h0fzfhnzoXnx2iXzVlXwOg3yOm3xOj3zOkPwuhPyy0-28EvxNwkwLgxwt3QVAORVxw-Tt0U2z0fmd3PyH3ozHyRhbEAJoWAKDTwKHnAL2kgIJmgKJlgJJngLJkQIpmQKR1QPQPNgPFv2b0TzZmYJTy-R80dwz2J09C-2HzU1XSrVQH-yVVoNJXoMQkYIYWYJHlYI1nYK1k4J1m4L1l4INn4Jwwv2N3Nh81EOwIKwkKNCkNQnfxlVQO-weQpyoInynyVG0M-jUIXzn2XznzXzn03znx3zn33znyPzn1PwULQLsPGxHR3gTxwIkJ8g7xPi7zey62LR8KlXkPikUMoL-2oLnwCLQCCJMRCLAPCLUMiLUOiLUNiLUPiLUMSLUOSPKNSPEyKO3jvxzWKE4LcO5VkNKL7xBAqJIJUMlHUKoU0M3XqKg10JDX0LxkML+WML1VMJAXMKcUsLBWsOj1sKGLRywMFxbwCW4MmKtQWP808PmOmPU1UOWKUlWOY3WLY02Ni22JbV2IQ32OjUOKBWOPjVOKw3OLP0ELSP5xuPN2cIf0FTyLxxnVFXFzJ2UJeNWzeNxL81q0+OqLUNqNQD+PVABMpSBLgxBO4zBN6whP6yhI5x6KQP6PcUuLuw5xGLEOyIfzdEeKLQ8IJIoOeJJKWLJJ+J9ypLn1CLUOaIZKKLaJgM6LZL1w5L4K5IH0GN5NswyNGOe0pPb1vAIJfyIL7jFJJ3IO8JHw+OlP8Pz1NJAMVLpMOxVPX3aJZLZ01IP16LhJSKEJqTT0cNuPEIfyPgxMIJK2WhtJSPtLxKlKVFlJoPl0CPSWpIi0X1axXwO3X06y3zL131ZwP3D2E3oJsP1IbyrQfSyNRPGNfXNJt2FStOggTIGKTOJPd1TKnwVJdO+CzLCLALzKiKgKLLiLgLLKSIuirIuJrJRzfx5QFMbJHPwNbIKIJwPHeKITIK8KUJ7IoTTIHKAOHKVNHIgILJiJgJLISIQIrMNNdIEJ5NrIyXrIkUFPGNcJjMtLjI1U7O5IlN3JC0+P7MaLPJAJzIiKvInNvOnIfLnNcXhNfKXNyxXKcPv3GNyJbM73x1FS6RAoq0zyAu7PK17JqInwzLqPPI9NzNgrYMnK6IQp4MfJdHnJQsXKEVKwzWNNmwmN-Nt3-J1UAr1LIrd2PPAppOn1ougpaIYqMKYs1NLMQopg4uDMRMW14tXKwqnI3LwqxK-WuKIqbg-0JNd1+zHykuospNksvJYOvI6Pgq4JnL6KQpQNQu4oBQ-MOTGPvP0vyPwq-XFSJPIoXVtIPMqLAqosHLsoKhVPzLgo4LvO1IsLYrqOQo0qGPpm0swr8rcoCsxPIxdUlNMteOAtCokpGRPIgroLiqYLHLVMYucrMNcv4INnUoGJDKqh8pk3c3oJFKvVErmIqoso2yspisgoYLkoSvHOauSpYrSvcpfK4t2TjwwojK-JoqfwtKEu7znVKvCsTMPLCuqustiqgvsoMMct9OLMWpOPSufOrO6tSV6qF1b2bLYk3KCqfmGrtJOqqvkmkvXLoOBxVMX2LxvLYLLy31Z130fIdD+PYu5yyoNJvzQCm02rXIvMKtjP2uRj+sisOuqvlxBq0LBtVzmoMKD2Yo4LhtnLJiRu2pWpepESbzypNNYMGvfOJv3PMuxXGr-1JpxtBpAMpqaupqgJhv8rMIRoyrnJRq6s0pMXj0-OxvYO5pkMquWzKNIoBssqFon220CIprAMhqcuhpgPpoKr1iZtso8tWqcwAvZqxt0otu5sIu1sl11rEv1sFs+OFo9On1NvivNt9PX2ltSqOLlufI6sVqAtZupVVt8s5rtG5pahMqOq7L9t-wDqNsHJDqYLDpps1NhrgJjrtqHIdsTuGJdpRLdt4O5uMq9qdxIt9qiqWMDuNrqMLr0OLqlqtvLrOkrvjr1JruuMmz4tb2FMErbOEsJ15rbpGvEoNrzoAO7spN7q2P7strpqHsZpryrpZuVss2Tr6oCQGtnq3OxOKIdL3KXv+o7tTK7oLrFrNogJLsjphNZIrsPtHpGprvTQxqnovujNwsCsMvMkJv5p-18MNvXtfoYPFqZM9M-sHq4N-oVurpPsNLPveovs+rkTxsKLT0zt-QipgZKMdOfvzqAK3sBJ3qMK-utvaqcRHuwaGLDOAZ0vypuDNK+oMuKvwzIb5q2BzrgdUL4blhuApMeFn1QBuEVJuGaJuFaJuHaJuE6JuG6JuF6JuF1LmI2BrvRseg5tmxkZ2u+sgeXJEYfs-zEafrQCkfDgsbfjkboUUZ2mUbVlUYOnUaOk0ZOm0bOl0Yun0Y8qMZwcDHDPrt4cIefz2sKJ3JbrkNmPPwceJvkmcaVVcZoXcaDU8YKm8b2l8YJn8aJkCZJmCbJlCYpnCZfMiaGJEKojMfc2Kc1tvuTLKoJIybIaybJNydJXycQkKaHmKZHlKY1nKa1kqZ1mqb1lqYNnqZsMaYNIcJaddt4Z-PAaKtf0cmgd6ZSdHw02yaoUGc3WGeCNGYYXGbxkmb+Wmb1VmZAXmacUWbBWWYuNWbfImw2diZNPKfTs6aPPIYUMObGtzslFOaUnOeY0uYaOuZDVuZbXuYQ0eejWeaBVefjXeaw0+ZQu+bQvu2RIbLdsqabuBdOtEZXv9qhYGdkfkcRdi2Rbg1Re43Rd60xf62xfZNxeE3xbBfHqNJ4YBbTqvp+qRNsZ9sMZpchaVGheJVhZ93hagyZcpRZcOzZdVI5fLy5bZx5a1L5afIFfKMJa8ocX5NaYCVqe5pxIhdSbMvBYFrlacfpZdJVbYzVZ1A1YNC1ZNB1YtD1ZtANYdCNZdBNe5LNbWtNxJbVrJZnp2eIYJwdyOdILsf7yddgeOckbdbccZaUZUbUY0a0Z0b0YMfSZrvzhidJfyvye5tIdTfvulfSdlYkbpekYZY8YLZ8aLYCZLZCbLYiZrq4dMc2YBbAYEYgaEfjMXubfsdbezahfArVY7EeHPI1e9cap1f9aYoNeDdcvqYNnyZWeMZHUzTHfMfid2rnvxso0bfxIoczaocWPleXZsrXY8bkq1d9ccr1cDZSqNdDceuPa+eHbPZAfQXafFesecylbSfnfEcXdfekfffXa-a3d-d3YA4PeA8ytNZruadHRFfMe2cnd2fbOdrg8dYXeoacbffdbQ68Yw78aw6CZw4uhA4JZrvWaI6tcg5wrI6TYIspaqupcQ9o4Ufo7ccY6KeY7KdY6qfY4pk48FZwd+d44vbaatCBYOZo5fbo5Q4Y8-aY58cw4Cew5Cdw6HZwZalys0+tbFcTb-PxubvtZmOo-E-08k8M+k+M9k9M5Y-M7Y8s447w8jaFberuMg-RKc8SYJxCrc7TbnYzb09AqXZ87yZk7Gbk6mYU7maU6PbC4HyjadtPo2v+fMYTcE+c8KLtedYddeKfbvrS+Q5cdQ78+y4C-k6C8U5C+U6K8McAfA+I-czreg+ndFNnfg5S885a4M7a6M4KfQ669y56-y768K+s6GKrfqRG4CQ-dxpq4JwbcS6bem6a66YoSVZbBXaVAO5GeW5KbM4qYs5qas4aeHZMfPYq7aavasYm7rKm488cYUeFo1dXeBy1e9fNr1f9cjqNeDZjo9ZdAO5PZwZMcxp++tb1HrZE51vO9S5TKcbB43seEh-Fp1d9ZpoNcDetsPacWR7QFR9A6iZHUx5rYBdI6IaO9FWSdO4fbBcJ4otB+kdJ-J68Zh+p-h7p6R5r2Z645wcI-Z7jd4YE+57i+8zx+9oJ9m6J5F5cbF48Yp8l78el6Cdl44--orZwZ4+V5TvMe0-G72a1V091+F5XaccN4KeN58al4CZl7OkZ9QHl9U6GPU7t-Psg8c+q416-QzvveIuS4u5BayZJ-dfF6KZN7KbN6qYt+U6t-sZrts+4b47Zhtad4o-eyB8a6F6u7T7cYz7Gaz6mZz7mbz6PYL4zYi7ro58q8scEed7K6o5r7d7r9F-T6N4l999N-9-N8D7l875K6vyRMnr2-QTG9i5vdq619bqT9r5GQ94Ua95GZ95Kb94qYD7JiD5D-w5waAdQAj-wfX4nfV63+TZ34a56f39T-H4b8n8z+n7Z9Z+ufefpby24GkduD-CDmzDJ798p2g-QHK7xB6H9wed3Rvjc2b4PNW+LzdvmChv7hcomGPaAfGGOD8NX+19L9AgmH4EkDgo-EZKQLljHAKSYgeRscEVLHBmixwVoscHaLHBOixwboscF6LHBy2n+PYKe0tb2d0ETAuAeR3np3t+eifabrQJB4MDw4Mgt+CwLoRsCdoHAtWFwIOg8CjofAk6AILOhCCLoIgjyuINZ6SCse0gv7gP0r481qBFDFQZk0JRqClUGgmhFoKDQ6CCoegvaAYIJhGCiYJgkmGYLJgWCKYVgl8jYKabCtS+JAnHhX3npa1FB3TNwd-08FkkfBpKPwYhACFDwghI8EIRrDCFawIhOsKIXrBiEGw4hNhBIWsySFSC2YIQj2h-3c6vF3BfTXIYwOYGsD2BnA7gbwP4GCDhBog-vM0J+atD7B7QtXgkzf6ip4+mQrOtyV6EJ9heXgqhPkM3SFDgixQhhKULxjlC-klQvVNUJAS1CnE9QsFI0IuIzCiWIlWNvb3cwRCKW0DTYWsPoF5DBh2g4YfoNGHGDxh5gyYdYO77VsVeJpKIdzQS71duhNAnIRph2FKQ9hzGA4Q0SOEhoThLaM4QhguHRorhQKG4fGjuFYYHhKFJ4eazK5Qi3hASGIbay6FJdlByI1QqiOJToifcmIqDNiNiy4i4M+I7jISN6zEj+spI9kuSOEyUiFC1I6NnyVeGR92hVXcgRK3WquDZRbIyUByJfhciOwPItjHyMpQCjDsQo1UiKPLxii2cEorUlKKfIyjyico0rrg0VFP82Yfg+tlQK2HrCB83whERJx1Etg9RSoA0eqCNE6gTRBoM0SaAtEWgrRNoG0Q6DtEugHRGwz7nMN77vCX+Swigb9Wr5Ii6B8kPUWgCNH6jzyJoiMY1QtEximKNohMa5TiEGw-BTQiQeV0zEMjHB8A5wQoP9FnczKforNgGPAqliQx5YuSmaKjGOUrRcYlKnaKTGPVmxjwsDpF0jLFAgh3NPnr2IF6OitRSoYsagBHFoAxAY43QVWKnG1jZxDYhcQN3PxOjl+zmOzvMJIFc8cxaoyjt6NBY7jCxng4cTZWPHaDxxZ4wwReNMFXiLoi4qkdx2G7JC1xiw69rmNKwaivxqg38S6X-H+DAJ+g88cYMvHmDrxEItTtBLaEkDHem-BCYtiQkbDdxJY1CZoJPGBCgJoQkCZELAkUwIJsoovkRKfFrjo+qomDpii+HUSDxtE3wfRJKGMSKhzEmoaxKbE3ixBkI3bjBJLExcY+yw4KsyL7E9ChJ+4w8agHQlFDMJwQ7CeENwnRD8J8QmujlUUnES1xKo18fxIVGUTfR2kkSQULEnHCJJ5wqSdcJklgp2JjoobiuK2p6SyB9kgHi6KclzEBxz7ObsJMYF-j3JOIzyQSO8kkjfJWGfyWmJwaQDvu7Y9BOhM9EaTtxVE78SiNcn7DEp-I5KcKNSnij0pwmTKb6PTFBTsaAQ7ml6J+F81opzXPXqWJLEb0xAwOM0RGPNpWiYxkdO0QmJjoGiXQ6ElsejzZ7EC1xqQsiW+MwKRTbx2k4WiaP1FDTxaFoqMTTRtFxjrajYpxDNKPGd87xGBAmj32hGzYOhaQ-GpuMHGaSCxqg7aQNL2m6DRpR0iaadOmk145pS4xXotLX4LDZBQnaQkVKUH9itpjAr6doP2m-TDB-00wYDPAlXSoJeDKLu0NImqTyJLvfMdkNKnsjPpaE76YEJRmhC0ZkQjGWxKxmEScZq4ksbxLCkIDVhW42GVpNJnajyZmgymSUOpkVDaZNQ+mU2MZlDFi+UA8GSQJUl8TwprnLmVkM1G8y9x-M3wYLOOHCzzhos64eLL8mSyDSE9EvjZJLF2T4Ja0hehtLEHwz1BiM-wcjP0F-TjBAMs6BdL0lGy3yVkmWUpJCmQyeeX6Orq9OKnOS1ZJYjWQUK1k4idZBIvWSSINkZSvZzwvXMzOCkFSnphRFNp1PTbdTLu9AyOfsOjn8jY5wo+OeKMTkNTk5NIl0avz9mDSA5sfWuIJPDkHjC5GI4ucaNLnmjy51oyuU+WBmQTCBYMv2RsFCmWyHJ36YmQoVOCtyx5csDYBST2DyMNgipDYM0Q2CtENg7RDYJ0Q2DdENgvRDYFMOOCti6RSo+MIvMblqTMo0DWeSD3nnhwr5b8ZeXQlXk7R15asTeQdG3lHRd5J0feWdEPkXRj5HlU+bYNdG4zL5nYuQc9Jhkqzyi98jwRpkflKpn5NCV+UGnfkFRP5e0b+QTF-lEx-5JMQBWTGAUUxQFL5cBYkLsF5S2YuCjpnfKEmoKqE6C0lJgsQjYKh4uCkePgo1iEKtYxCnWKQr1jkKDYlCmwtQpaG0L7p7mfBZ0KYVzyySbCzdBwuCJcKGEPCvGHwr+QCK9UQikBCIqcRiKwUEii4lItmEyL6R6CQhTp2nmILmFyipeSvLXkbyt5O8veQfKPknzOJVii+cUGIWfD7F3JJBX0JQVOKXSaihohopDRaKW0OihDHoujQGKgURi+NCYqwxmKUKFilOSbNHbcSMabMieeFPhEhzuZBJUJR+PkgsKlIKi5jFEqgwxLYscSuDAku4xJLesKS-rGkvZIZLhMWSmeZZIzGyKAk5CpkYoofkRKX5Lij+W4p-keKAFXikBT4rv7DLrF9Ci2f9wQHZzlZPouYpUp+HVKplGCmZTgrmUEKFlJCpZRQpWXbc1l-itAJgsKkTLkFqhGpcSjqU+4GlbGJpZShaWHY2lqpDpeXi6Vs4elWpPpU+QGWILmpkClmQ-2zHFKEBHU3ZZ+JCXMLwKvyjsHsHPL-KdQgKg0MCpNCgqLQ4Km0JCodCUKDYmCyRWfOskFKH+MCqGbfOCUD4DlqK6pZipso4q35clAlRjUcokrClAC1ylSrBQ0rzFy4tsSMpsUrSCZVsl6TFPKUUN2VZS4Xp8pbBYqlQPKrBXysapEqMaTFMlRjVFWPUJV2SgjlxLoWXyXxSK5wRkNRV81VVSq9VVysiW4q9VX8wVUapSoUqMaZquSf3hyU1yeOuUmVfQrglbLnBntHOcl2dU9TXVC87lR6o-n6rvVf831aaoujmrBlTM6Vessvn4yFZHM+BXsvPzxr85nKpNe6t5WpqvVP8n1SKqAUBqwFvi-NQ8of5FKo189JWWqrLWf4K1KfQlBqoxrJra1OCtNQ2ozVNqyFLaqhQpI04Mqxlmc+LqWrRVsqMV1al+SmonX1qCFjakhVmopg5qYVODH2WGoLUBLNlTg+esHJdX9qg1m6p+WOt1V1q8F6aohZmubXZrA1wa+UanPbVuj4wTyldaKh2V9r11+yp9WgpfWcLPV76qdZ+pnWiK51tK7KVavDXAbEV3a29iiog1OroNrC2Deovg28KP1gir9bOp-Wtrh5LUt2tgvalrqCNc84Wv8uxXA5AV+K82qCuJWR1IV5KmOt8pdA6q0NQxIgbLICVyri13Ypjem0HWnVqlrGjensA43i1gVhKmmuCtJXW0xV8aITY8s75-rnR0TU2UuttU4akmsmuNcwqU2RLVNH87jZpr406bBNNeETZKtBlpzsati0DdDJeVhK3ltml+fZpwWOaf5zmgBa5uzWGbsZd0y9YauvmEyk6rKqDSxoXnKaQt3CsLQQoi0kKotx6mLXmrrlmzO1iWq2ZzPw1yabN6WuzW-LU3Zb+FuW4RflupWFapZI8krcutWmTze196yDeWuq1PyMtdWhzV-Kc1-yXNZ0fTagHc0WqcGeSx-lAqvVlbJ5pSvrcxoflBaMFmWzRQ1t0VNbDFLW8VW1oNI+zFt8KkDd1vCl3qE1D6+TYDWHVbb2FO22JXtsSUHbUlR2rDLNtzVDF7+524KTqu5rgb1tVWtLUNtq1YL6tY28LRNsi1Ta3NJ2t8pAIB3Y0VNK28KSd0dVg7NtNW4LSNtC0w6ctcOvLQjui00bOG4mv2ePIs0E48NoO5LpcFblkkKSNmdUIqU4EHQNGJ0bor0SmFiA6V+S61cxAx0ICexlWxnUJJZ0uk2dGNHaJzoJjc6SYvOuch5QF0QLz5QGgJaLucGKrbt-Wz-EzpB7S7DEK8+XfoK50BMedCBVXS+XV00K4VgOnXekKs3TcjdryyUCbomRm6CoCujWErp1gq61Kau2LZrqW0ljnd+NGNdjsl3M79EMun3UPD91-IA9ICIPR1RD1Fa-mwuo8ZHsKIVaGdbuqXfHtN1vzzdwQy3RU2t08FbdNhe3QaWlmZFMNxQItezOcG9b9dfNd3QFs90l7vdZe33RbsV1W7ldNu4PXboXVN74twfPPauugbd6qlhKL3XWkT0MJk9CGVPUCnT1gp+dQyvxVrrnyz6b68+4vd3AT0D6k9Q+-3SPsD1j6M9E+1ZfvvD2Ukj99uV3WZQX2HKl9felfRfrX1X6U9N+tPXfp32Z67lT++FeXF83NyUt5+T-Ryu-1n7S9WC8vWUMr0zNq9FhWvRcXr1vkR2U+jtdhpvX416dne9NvAYg26IF5Mu88uOLVi-sjoKVVyo9TNQKFcDKcr7ktLl2Hcm5NjD8V3qEmYqaDdCOg3tAYNEwmDw9TOIGvYM1yTNi6nPTNtf3EEbZ-eCg31qoNPzhDQaUQyPHENaxJDjNaQ2AYNKEcL1Ha8zcQcKIOqJdRe1uUIbeQiHdB9BvxowbgJSG2UMh0PfSsUORqrDBOaPbYY-2CHqDjhnQ84bEOuGJD7how54ZMOWLANz+1vXavnoF6yDsekHg4YDROHAhLhspm4a4IeGIEXhmzhhun1dr-DoqDvfnIEP2HQj2R8I7kciP5HojhR2I8UfiO5KyjHa+WW3vnprb0jdhzI-UfAw5GSheRqZgUbMJFHE0JR7Kt0YP0v6eDN8nqrAcN0hGtDYRxCLobxj6G9Uhh22sYYf1-aFjz+qA1du2Xv7Xi6h-XZobQXaHtjERvQ1EYMMxHDjcR44xANOOQGiDXY+QaQZqPkGNj9xrY8ER2Mto9j0aA42wyON17YVj4xQ0ysDl5jVDNxytcOrlgb1lchwnaDDwOhf0ToMdBHKsE6M1yqdJWqTX0dvZXGCSaJodSgsxMy6hpuJnxviZgKEm1kGyUk-+vkOo63alhv43ApP1zzGThiZkwVDxMEwCTJMIkzXijwoVZD-6pXlwb0nKGwkaxtQ8wtFMTJxTQ8SUxrGlM6xZTXJz42+Vt4qnkjtO4TsKYfnam60uphhPqb+SGmQExpyPNyedHh8VTFRwU-nppMUM6TCmjE+HCxMOmQ0TphDC6aBRuneEHp+8cS2K0MrejKRlzv6bYNamQzTJ7QSyZKZsmOCHJllCabhPzaOtDK69b6bn0anAzD2hk5mbFPZmJTrJqU+yZlOcn3TpplOWdpVPnH5Vk8m7YCYyMe6lQwtUMw2b1NNmDTLZo022djMdma5-27s78dgVZy0z5Ras6vUlAjmsz-gnMxMzzMmECzPaIszgcralnFDyeaA9aSrMZmlUo5nc42dzPNn8zrZws+2eLOU7FpAAXSAiYA7ASAWQBADJB2AEARgUAFAB0BQAGAggNQAtAYAIBKAjqSgBKgAAEAAamQsAByMEBhbQvIXSgcFhC1fSQt-QlASAAwIWAgAwBzAygGgL+DYB2Augn5pQJgDoB5BWQZgSgC4EKCCB0C2QAAO40BlA9AR5RlHgA0AyAWAV1jKGgsuB7AoFkAPkCgBsAaLsgMgI+GDDFVEQzFlwDgAGAABHBgP+ZYvqAWLpAbIMZbYvoBEQQEFwD9QKXaXdLyQAy0ZboBqBaLXgZi6xYGC2WlA0AHQMUFAAAAvewOSGsA6AIAcAToEBCUtpAGLkDVQJReou0Xag9F8yzpYGCyA2ACAKwDoGyB5AyA+QLMKAGgtMAEgAAGRoDZXMAIl5EEBAgC-m2AAAazgBtAyLMAOy0RdxQOWMrWVnK4IC6BAQZLcl1AMYBADIA7AjV5IMIFkBeBjwEVvIFABYtxRirSAUq-1YpCYB0ryQMIKpbYtKAZrSARwPVbQC-m9ASgKJAgDYAMAIrbAEINkEUvKWrAZAci4YEPBMWFLCAHAJteSCZWZr2QKAHkEOvJBLr11uAFdeqtRWlA41ya9mDYBWBqrSgB6ypbUsUg2AOAJAFADoC2WKQSl2QDRaWuixQA81uAItfBglWEg2QZpEBACS1XOg356iL+f-OAX6LIFkaxmEECYBhrRNhICTYJtTXVrFNqK+9aGtxXRr0Nx8JtccvoA7AUFjyyABiupX0AZAP87ZaAhI2nr6ln6lpZACS39Lhl2QMZbctmWorx4dW6pYltwB4Q1Vja1tfQDOWDbrl0y3LaIIFKgIGlhTNrd1tOX9bhtp2xWGxvFV7LOt22yAHtu+33LA1pQCLfkvi2bbUtnW5bYRsgAdA4l25HkGsBJ3lAwYHyNjf4BiA1bsgJS8jcfAK24roAL2+gEyvZWCg-VhkDXAStUWaLdFhiybe-Nt2lAuNuICjeABRWgAA)

Reply:
Thanks for the clarification! 
I guess my follow-up question is similar to my comment regarding `--from-config`, if this is not a strict requirement for getting a working version of flexible plots, should it be included in the P.R.?

Reply:
Yes, as its currently used in `_adjust_vega_renderers` in `dvc/commands/plots`. Also, if we would not be using this field now, we would have to have separate logic for `VegaConverter` for `vscode` and `DVC`. This field allows us to modify the datapoints we need in `commands/plots` depending on whether we produce HTML or JSON.